sql/core/src/main/scala/org/apache/spark/sql/execution/SortExec.scala

```diff
@@ -124,9 +124,11 @@ case class SortExec(
   // Name of sorter variable used in codegen.
   private var sorterVariable: String = _
 
-  // The result rows come from the sort buffer, so this operator doesn't need to copy its result
-  // even if its child does.
-  override def needCopyResult: Boolean = false
+  private var _needCopyResult: Option[Boolean] = None
```

Comment:
do you mean `doProduce` may not be called?

Reply:
my bad, I rechecked again and I found we didn't need the change in `SortExec` and `HashAggregateExec`. I'll update soon.